app/workers/concerns/code_analysis_mixin.rb

```diff
@@ -19,11 +28,52 @@ def merged_linter_results
     results
   end
 
+  # run linters via pronto and return the pronto result
+  def pronto_result
+    Pronto::GemNames.new.to_a.each { |gem_name| require "pronto/#{gem_name}" }
+
+    p_result = nil
+
+    # temporary solution for: download repo, obtain changes, get pronto result about changes
+    Dir.mktmpdir do |dir|
+      FileUtils.copy_entry(@branch.repo.path.to_s, dir)
+      repo = Pronto::Git::Repository.new(dir)
+      rg = repo.instance_variable_get(:@repo)
+      rg.fetch('origin', @branch.name.sub(/^prs/, 'pull'))
+      rg.checkout('FETCH_HEAD')
+      rg.reset('HEAD', :hard)
```

Comment:
Pinged Brandon regarding this already, but I think we should try and get this fixed instead of doing what is done here:

https://github.com/ManageIQ/miq_bot/pull/473#discussion_r384919356

This would save us a few lines of code, but also ensure we are doing it in the same fashion in all places that need to perform a fetch.